Code/GraphMol/ChemTransforms/testChemTransforms.cpp

```diff
@@ -614,6 +651,76 @@ void testReplaceCore() {
   BOOST_LOG(rdInfoLog) << "\tdone" << std::endl;
 }
 
+struct CoreTest {
+  const char * smiles;
+  const char * smarts;
+  bool replaceDummies;
+  bool labelByIndex;
+  bool requireDummyMatch;
+  bool useChirality;
+  const char * expected;
+};
+  
+void testReplaceCore2() {
+  const CoreTest tests[] = {
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", false, false, false, false, "[1*]OC.[2*]NC"},
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", false, false, false, true,  "[1*]NC.[2*]OC"},
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", false, true,  false, false, "[3*]OC.[4*]NC"},
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", false, true,  false, true,  "[3*]NC.[4*]OC"},
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", true,  false, false, false, "[1*]C.[2*]C"},
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", true,  false, false, true,  "[1*]C.[2*]C"},
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", true,  true,  false, false, "[3*]C.[4*]C"},
+    {"C1O[C@@]1(OC)NC", "C1O[C@]1(*)*", true,  true,  false, true,  "[3*]C.[4*]C"},
+    
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", false, false, false, false, "[1*]OC.[2*]NC"},
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", false, false, false, true,  "[1*]OC.[2*]NC"},
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", false, true,  false, false, "[3*]OC.[4*]NC"},
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", false, true,  false, true,  "[3*]OC.[4*]NC"},
+    
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", true,  false, false, false, "[1*]C.[2*]C"},
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", true,  false, false, true,  "[1*]C.[2*]C"},
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", true,  true,  false, false, "[3*]C.[4*]C"},
+    {"C1O[C@]1(OC)NC",  "C1O[C@]1(*)*", true,  true,  false, true,  "[3*]C.[4*]C"},
+
+    {"C1O[C@@]1(OC)NCC", "C1O[C@]1(*)*", true,  false, false, false, "[1*]C.[2*]CC"},
+    {"C1O[C@@]1(OC)NCC", "C1O[C@]1(*)*", true,  false, false, true,  "[1*]C.[2*]CC"},  // seems broken?
```

Comment:
I would rather these return different results, but it seems rather hard to do since it appears to be a second order sort, yuck.